File: user/src/main/java/com/baigon/user/service/LogService.java
// 百工谱 — 日志服务
// 封装写 logs 表：info / warning / error 三个级别入口。
// 日志写入失败只记录 error，不抛出——日志不应阻断业务主流程。

package com.baigon.user.service;

import cn.hutool.core.lang.Snowflake;
import com.baigon.user.entity.Log;
import com.baigon.user.repository.LogRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

import java.time.OffsetDateTime;

@Service
public class LogService {

    private static final Logger log = LoggerFactory.getLogger(LogService.class);

    private final LogRepository logRepository;
    private final Snowflake snowflake;

    public LogService(LogRepository logRepository, Snowflake snowflake) {
        this.logRepository = logRepository;
        this.snowflake = snowflake;
    }

    /** 记录 INFO 级日志 */
    public void info(Long traceId, Long userId, String userName, String userIp, String detail) {
        write(traceId, userId, userName, userIp, Log.Level.INFO, null, null, null, detail);
    }

    /** 记录 WARNING 级日志 */
    public void warning(Long traceId, Long userId, String userName, String userIp, String errorMsg, String detail) {
        write(traceId, userId, userName, userIp, Log.Level.WARNING, null, null, errorMsg, detail);
    }

    /** 记录 ERROR 级日志 */
    public void error(Long traceId, Long userId, String userName, String userIp, String errorMsg, String detail) {
        write(traceId, userId, userName, userIp, Log.Level.ERROR, null, null, errorMsg, detail);
    }

    /** 统一写入入口：失败仅记 error，不抛出 */
    private void write(Long traceId, Long userId, String userName, String userIp,
                       Log.Level level, Log.RequestMethod requestMethod, String requestUrl,
                       String errorMsg, String detail) {
        try {
            Log entity = new Log();
            entity.setId(snowflake.nextId());
            entity.setTraceId(traceId);
            entity.setUserId(userId);
            entity.setUserName(userName);
            entity.setUserIp(userIp);
            entity.setLevel(level);
            entity.setRequestMethod(requestMethod);
            entity.setRequestUrl(requestUrl);
            entity.setErrorMsg(errorMsg);
            entity.setDetail(detail);
            OffsetDateTime now = OffsetDateTime.now();
            entity.setCreatedAt(now);
            entity.setUpdatedAt(now);
            logRepository.save(entity);
        } catch (Exception e) {
            // 日志写入失败不应影响业务主流程
            log.error("写入 logs 表失败（已忽略）", e);
        }
    }
}

File: crawler/src/service/log_service.py
# 百工谱 — 日志服务
# 封装写 logs 表：提供 info / warning / error 三个级别入口。
# 日志写入失败只记录 error，绝不抛出——日志不应阻断业务主流程。

import logging

from src.repository.log import LEVEL_ERROR, LEVEL_INFO, LEVEL_WARNING, LogRepository

logger = logging.getLogger(__name__)


class LogService:
    """业务日志服务（写入 logs 表）"""

    def __init__(self, log_repository: LogRepository):
        self._repo = log_repository

    def info(
        self,
        *,
        trace_id: int | None,
        user_id: int,
        user_name: str,
        user_ip: str | None,
        request_method: str | None = None,
        request_url: str | None = None,
        detail: str | None = None,
    ) -> None:
        """记录 INFO 级日志"""
        self._record(
            level=LEVEL_INFO, trace_id=trace_id, user_id=user_id, user_name=user_name,
            user_ip=user_ip, request_method=request_method, request_url=request_url,
            error_msg=None, detail=detail,
        )

    def warning(
        self,
        *,
        trace_id: int | None,
        user_id: int,
        user_name: str,
        user_ip: str | None,
        request_method: str | None = None,
        request_url: str | None = None,
        error_msg: str | None = None,
        detail: str | None = None,
    ) -> None:
        """记录 WARNING 级日志"""
        self._record(
            level=LEVEL_WARNING, trace_id=trace_id, user_id=user_id, user_name=user_name,
            user_ip=user_ip, request_method=request_method, request_url=request_url,
            error_msg=error_msg, detail=detail,
        )

    def error(
        self,
        *,
        trace_id: int | None,
        user_id: int,
        user_name: str,
        user_ip: str | None,
        request_method: str | None = None,
        request_url: str | None = None,
        error_msg: str | None = None,
        detail: str | None = None,
    ) -> None:
        """记录 ERROR 级日志"""
        self._record(
            level=LEVEL_ERROR, trace_id=trace_id, user_id=user_id, user_name=user_name,
            user_ip=user_ip, request_method=request_method, request_url=request_url,
            error_msg=error_msg, detail=detail,
        )

    def _record(self, **kwargs) -> None:
        """统一写入入口：失败仅记 error，不抛出"""
        try:
            self._repo.insert(**kwargs)
        except Exception:
            logger.exception("写入 logs 表失败（已忽略，不影响业务）")
